lib/msal-browser/src/cache/TokenCache.ts

```diff
@@ -240,40 +241,43 @@ export class TokenCache implements ITokenCache {
         clientInfo?: string,
         requestHomeAccountId?: string
     ): AccountEntity {
-        let homeAccountId;
-        if (requestHomeAccountId) {
-            homeAccountId = requestHomeAccountId;
-        } else if (authority.authorityType !== undefined && clientInfo) {
-            homeAccountId = AccountEntity.generateHomeAccountId(
-                clientInfo,
-                authority.authorityType,
-                this.logger,
-                this.cryptoObj,
-                idTokenClaims
-            );
-        }
+        if (this.isBrowserEnvironment) {
+            this.logger.verbose("TokenCache - loading account");
+            let homeAccountId;
+            if (requestHomeAccountId) {
+                homeAccountId = requestHomeAccountId;
+            } else if (authority.authorityType !== undefined && clientInfo) {
+                homeAccountId = AccountEntity.generateHomeAccountId(
+                    clientInfo,
+                    authority.authorityType,
+                    this.logger,
+                    this.cryptoObj,
+                    idTokenClaims
+                );
+            }
 
-        if (!homeAccountId) {
-            throw createBrowserAuthError(
-                BrowserAuthErrorCodes.unableToLoadToken
-            );
-        }
+            if (!homeAccountId) {
+                throw createBrowserAuthError(
+                    BrowserAuthErrorCodes.unableToLoadToken
+                );
+            }
+            const claimsTenantId = idTokenClaims.tid;
 
-        const accountEntity = AccountEntity.createAccount(
-            {
+            const cachedAccount = buildAccountToCache(
+                this.storage,
+                authority,
                 homeAccountId,
-                idTokenClaims: idTokenClaims,
+                idTokenClaims,
+                this.cryptoObj,
```

Comment:
Is this only needed for the base64Decode function? Can we pass that function in directly instead?